source/MultiGrid/MultiGrid.js

```diff
@@ -0,0 +1,338 @@
+    styleTopLeftGrid: PropTypes.object.isRequired,
+    styleTopRightGrid: PropTypes.object.isRequired
+  };
+
+  static defaultProps = {
+    fixedColumnCount: 0,
+    fixedRowCount: 0,
+    style: {},
+    styleBottomLeftGrid: {},
+    styleBottomRightGrid: {},
+    styleTopLeftGrid: {},
+    styleTopRightGrid: {}
+  };
+
+  constructor (props, context) {
+    super(props, context)
+
+    this.state = {
+      scrollLeft: 0,
+      scrollTop: 0
+    }
+
+    this._cellRendererBottomLeftGrid = this._cellRendererBottomLeftGrid.bind(this)
+    this._cellRendererBottomRightGrid = this._cellRendererBottomRightGrid.bind(this)
+    this._cellRendererTopRightGrid = this._cellRendererTopRightGrid.bind(this)
+    this._columnWidthRightGrid = this._columnWidthRightGrid.bind(this)
+    this._onScrollHorizontal = this._onScrollHorizontal.bind(this)
+    this._onScrollVertical = this._onScrollVertical.bind(this)
+    this._rowHeightBottomGrid = this._rowHeightBottomGrid.bind(this)
+  }
+
+  componentWillUpdate (nextProps, nextState) {
+    const { columnWidth, fixedColumnCount, fixedRowCount, rowHeight } = this.props
+
+    if (
+      columnWidth !== nextProps.columnWidth ||
+      fixedColumnCount !== nextProps.fixedColumnCount
+    ) {
+      this._leftGridWidth = null
+    }
+
+    if (
+      fixedRowCount !== nextProps.fixedRowCount ||
+      rowHeight !== nextProps.rowHeight
+    ) {
+      this._topGridHeight = null
+    }
+  }
+
+  render () {
+    const { height, style, width } = this.props
+    const { scrollLeft, scrollTop } = this.state
+
+    return (
+      <div
+        style={{
+          height,
+          width,
+          ...style
+        }}
+      >
+        <div
+          style={{
+            height: this._getTopGridHeight(),
+            position: 'relative',
+            width
+          }}
+        >
+          {this._renderTopLeftGrid()}
+          {this._renderTopRightGrid({ scrollLeft })}
+        </div>
+        <div
+          onScroll={this._onScrollVertical}
+          ref={(ref) => {
+            this._scrollingContainer = ref
+          }}
+          style={{
+            height: height - this._getTopGridHeight(),
+            overflowY: 'auto',
+            position: 'relative',
+            width
+          }}
+        >
+          {this._renderBottomLeftGrid({ scrollTop })}
+          {this._renderBottomRightGrid({ scrollTop })}
+        </div>
+      </div>
+    )
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState)
+  }
+
+  _cellRendererBottomLeftGrid ({ rowIndex, ...rest }) {
+    const { cellRenderer, fixedRowCount } = this.props
+
+    return cellRenderer({
+      ...rest,
+      rowIndex: rowIndex + fixedRowCount
+    })
+  }
+
+  _cellRendererBottomRightGrid ({ columnIndex, rowIndex, ...rest }) {
+    const { cellRenderer, fixedRowCount } = this.props
+
+    return cellRenderer({
+      ...rest,
+      columnIndex: columnIndex + columnIndex,
```

Comment:
Guh, this is a typo. Supposed to be `fixedColumnCount + columnIndex`. I will add to the list above.